client/a8c-for-agencies/sections/sites/features/a4a/site-details.tsx

```diff
@@ -44,17 +51,20 @@ export default function SiteDetails( { site }: any ) {
 		);
 	};
 
-	const onRemoveTag = ( toRemove: string ) => {
+	const onSetTags = ( siteTags: string[] ) => {
 		setIsLoading( true );
-		mutate(
+		mutateTags(
 			{
 				siteId,
-				tags: tags.filter( ( tag: string ) => tag !== toRemove ),
+				tags: siteTags,
 			},
 			{
 				onSuccess: ( data ) => {
 					setTags( data.map( ( tag: SiteTagType ) => tag.label ) );
 					setIsLoading( false );
+					queryClient.invalidateQueries( {
+						queryKey: [ 'jetpack-agency-dashboard-sites' ],
+					} );
```

Comment:
Because we ignore subsequent changes to the tags prop we have to do these state changes here too. Ideally this is how the state would work:
1. Global state comes from `sites = useSitesQuery()` (or whatever it's called).
2. A specific site's state is eventually passed down to SiteDetails.
3. SiteDetails passes `tags={site.a4a_site_tags}` to AgencySiteTags.
4. AgencySiteTags does NOT keep local state, it only triggers an onChange() handler with the new value.
5. SiteDetails triggers the mutation in its onChange handler.
6. The mutation has an optimistic update and mutates the current tags state via `queryClient.setQueryData()` so the UI update is instantaneous.
7. The mutation uses invalidateQueries so the global state changes and we go back to step 1. Whether an error happens or the state changes while we're waiting for the mutation it will all be fine because ultimately we only load the state from the API itself without trying to keep a local state copy in sync inside components.

This way there's no local state tracking needed, everything is based on the API responses, and if something changes in between actions the change will be reflected (because AgencySiteTags will no longer ignore `tags` prop changes after the first render).

(obviously this applies to notes as well)